### PR TITLE
Add abstracted synchronous WebviewWidget that uses WebKit or WebEngine

### PR DESCRIPTION
**Pull request by kernc**
This seems to work with our Highcharts example on Qt 4.8 and 5.3 (Qt WebKit) an Qt 5.7 (Qt WebEngine). It's an attempt at a somewhat uniform synchronous webview interface.

Perhaps @PrimozGodec can confirm this works with his widgets?

Please review.

### CODE REVIEW

File: Orange/widgets/utils/webview.py [hunk 1]
```diff
@@ -92,31 +183,230 @@ def runJavaScript(self, javascript, resultCallback=None):
...
+        _to_be_exposed = None
+
+        def evalJS(self, code):
+            while not self._is_objects_exposed:
+                qApp.processEvents()
+            self.runJavaScript(code, lambda result: setattr(self, '_result', result))
+            while self._result is self._NOTSET:
+                qApp.processEvents()
+            result, self._result = self._result, self._NOTSET
+            return result
```
Comment (ales-erjavec) [inline Orange/widgets/utils/webview.py:467]: At the very least this needs to guard against reentry. It also needs to document that it calls `qApp.processEvents`. Similar for all other methods that call `qApp.processEvents`

**Comment by kernc** [comment]
I think I managed something. It's much simpler now. I introduced a slight regression with Highcharts, visible with only one of Timeseries widgets, but that's independent of WebEngine as it appears on WebKit also. Should find and squash it promptly. Please have another look.

@PrimozGodec you also give it another test, please.

**Comment by PrimozGodec** [comment]
Works fine for me.

**Comment by ales-erjavec** [comment]
Can you please rebase.

**Comment by kernc** [comment]
Done.

File: Orange/widgets/webview.py [hunk 2]
```diff
@@ -1,91 +1,3 @@
...
-        self.setContent(html.encode('utf-8'), 'text/html', base_url)
-
-    def svg(self):
-        """ Return SVG string of the first SVG element on the page, or
-        raise ValueError if not any. """
-        html = self.frame.toHtml()
-        return html[html.index('<svg '):html.index('</svg>') + 5]
+# This was once here. For compatibility:
+# pylint: disable=unused-import
+from Orange.widgets.utils.webview import WebviewWidget as WebView
```
Comment (ales-erjavec) [inline Orange/widgets/webview.py]: Is this intended to preserve full source compatibility? The GeoMap widget in orange3-text does not seem to layout correctly with this change applied.

Comment (kernc) [reply Orange/widgets/webview.py]: Gah! It was meant to, except in the two constructor arguments bit. [Word Cloud had been prepared for it](https://github.com/biolab/orange3-text/pull/99/files), but I forgot about Geo Map (fix in https://github.com/biolab/orange3-text/pull/115). Thanks for checking. Have now removed this file altogether.

**Pull request merged**